rotkehlchen/chain/optimism/modules/velodrome/velodrome_cache.py

```diff
@@ -0,0 +1,218 @@
+                blockchain=SupportedBlockchain.OPTIMISM,
+            ))
+        try:
+            db_addressbook.add_addressbook_entries(
+                write_cursor=write_cursor,
+                entries=addresbook_entries,
+            )
+        except InputError as e:
+            log.debug(
+                f'Velodrome address book names for pool {pool.pool_address} were not added. '
+                f'Probably names were added by the user earlier. {e!s}')
+
+        globaldb_set_general_cache_values(
+            write_cursor=write_cursor,
+            key_parts=(CacheType.VELODROME_POOL_ADDRESS,),
+            values=[pool.pool_address],
+        )
+        if pool.gauge_address is not None:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(CacheType.VELODROME_GAUGE_ADDRESS, pool.pool_address),
+                value=pool.gauge_address,
+            )
+
+
+def read_velodrome_pools_and_gauges_from_cache() -> tuple[set[ChecksumEvmAddress], set[ChecksumEvmAddress]]:  # noqa: E501
+    """
+    Reads globaldb cache and returns:
+    - A set of all known addresses of velodrome pools.
+    - A set of all known addresses of velodrome gauges.
+
+    Doesn't raise anything unless cache entries were inserted incorrectly.
+    """
+    gauge_addresses = set()
+    pool_addresses = set()
+    with GlobalDBHandler().conn.read_ctx() as cursor:
+        str_pool_addresses = globaldb_get_general_cache_values(cursor=cursor, key_parts=(CacheType.VELODROME_POOL_ADDRESS,))  # noqa: E501
+        for pool_address in str_pool_addresses:
+            gauge_address = globaldb_get_unique_cache_value(cursor=cursor, key_parts=(CacheType.VELODROME_GAUGE_ADDRESS, pool_address))  # noqa: E501
+            if gauge_address is None:
+                continue
+            gauge_addresses.add(string_to_evm_address(gauge_address))
+            pool_addresses.add(string_to_evm_address(pool_address))
+
+    return pool_addresses, gauge_addresses
+
+
+def query_velodrome_pools_from_chain(
+        inquirer: 'OptimismInquirer',
+        existing_pools: list[ChecksumEvmAddress],
+) -> Optional[list[VelodromePoolData]]:
+    """
+    Queries velodrome data from chain from the Velodrome Finance LP Sugar v2 contract
+    (Find more details for the contract here: https://github.com/velodrome-finance/sugar).
+
+    May raise:
+    - RemoteError if there is an error connecting with the remote source of data
+    """
+    data_contract = inquirer.contracts.contract(VELODROME_SUGAR_V2_CONTRACT)
+    pool_data = data_contract.call(
+        node_inquirer=inquirer,
+        method_name='all',
+        arguments=[1000, 0, '0x48a264d1BC1765598D327bB60789bEd541b12504'],  # limit, offset, address  # no need to use a specific address here we just want the pool data  # noqa: E501
+    )
+
+    pools = []
+    for pool in pool_data:
+        try:
+            pool_address = deserialize_evm_address(pool[0])
```

Comment:
the addresses already come checksummed... isn't that the case? If not you would need to catch the possible error here but I believe you only need `string_to_evm_address`. The issue with `deserialize_evm_address` is taht is expensive to call

Reply:
No they are not checksummed so we need these deserialize_evm_address calls. What error you refer to? There is already an except statement for catching DeserializationError.